fast64_internal/f3d/f3d_gbi.py

```diff
@@ -2711,6 +2752,133 @@ def save_binary(self, romfile, f3d, segments):
+        tile_scroll_tex1 = mat.scrollData.tile_scroll_tex1
+        if tile_scroll_tex0.s or tile_scroll_tex0.t or tile_scroll_tex1.s or tile_scroll_tex1.t:
+            bleed_scroll = True
+        else:
+            bleed_scroll = False
+        if LastMat:
+            # bleed cmds if matching tile has duplicate cmds
+            bled_tex = []
+            for j, (LastTex, TexCmds) in enumerate(zip(LastMat.textures, mat.textures)):
+                # deep copy breaks on Image objects so I will only copy the levels needed
+                commands_bled = copy.copy(TexCmds)
+                commands_bled.commands = copy.copy(TexCmds.commands)  # copy the commands also
+                LastList = LastTex.commands
+                # eliminate set tex images
+                set_tex = (c for c in TexCmds.commands if type(c) == DPSetTextureImage)
+                removed_tex = [c for c in set_tex if c in LastList]  # needs to be a list to check "in" multiple times
+                rm_load = None  # flag to elim loads once
+                # set tile size cmds need to be removed and placed in dynamic data if there is a scroll
+                tile_size = [c for c in TexCmds.commands if type(c) == DPSetTileSize]
+                for j, cmd in enumerate(TexCmds.commands):
+                    # remove set tex explicitly
+                    if cmd in removed_tex:
+                        commands_bled.commands.remove(cmd)
+                        rm_load = True
+                        continue
+                    if rm_load and type(cmd) in (DPLoadTLUTCmd, DPLoadTile, DPLoadBlock):
+                        commands_bled.commands.remove(cmd)
+                        rm_load = None
+                        continue
+                # now eval as normal conditionals
+                iter_cmds = copy.copy(commands_bled.commands)  # need extra list to iterate with
+                for j, cmd in enumerate(iter_cmds):
+                    if cmd.bleed(LastList, commands_bled.commands, j):
+                        if cmd in LastList:
+                            commands_bled.commands.remove(cmd)
+                bled_tex.append(commands_bled)
+        else:
+            bled_tex = mat.textures
+        return bled_tex, bleed_scroll
+
+    def bleed_mat(self, mat, LastMat):
+        if LastMat:
+            GfxList = mat.material
+            # deep copy breaks on Image objects so I will only copy the levels needed
+            commands_bled = copy.copy(GfxList)
+            commands_bled.commands = copy.copy(GfxList.commands)  # copy the commands also
+            LastList = LastMat.material.commands
+            cnt = 0
+            for j, cmd in enumerate(GfxList.commands):
+                if cmd.bleed(LastList, commands_bled.commands, j - cnt):
+                    if cmd in LastList:
+                        commands_bled.commands.pop(j - cnt)  # list gets smaller as I pop, so modify index by num popped
+                        cnt += 1
```

Comment:
I did this because I was worried about duplicate cmds within the list being accidentally removed. An example is a sync, which can be present in the same GfxList multiple times. I may want to remove just one of them, but the second one only: a case where I would have to remove by `list.pop(index)`.
This convention would also result in me having to keep track of the amount of removed indices to properly reference the correct one.

Reply:
Maybe you can just replace the command with a `None`, then filter out all the `None`s in the array at the end? That way the command index lines up between all the display lists.